src/plugins/vis_type_table/public/components/table_vis_basic.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { memo, useCallback, useMemo } from 'react';
+import { EuiDataGrid, EuiDataGridProps, EuiDataGridSorting, EuiTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { orderBy } from 'lodash';
+
+import { IInterpreterRenderHandlers } from 'src/plugins/expressions';
+import { createTableVisCell } from './table_vis_cell';
+import { Table } from '../table_vis_response_handler';
+import { TableVisConfig, TableVisUseUiStateProps } from '../types';
+import { useFormattedColumnsAndRows, usePagination } from '../utils';
+import { TableVisControls } from './table_vis_controls';
+import { createGridColumns } from './table_vis_columns';
+
+interface TableVisBasicProps {
+  fireEvent: IInterpreterRenderHandlers['event'];
+  table: Table;
+  visConfig: TableVisConfig;
+  title?: string;
+  uiStateProps: TableVisUseUiStateProps;
+}
+
+export const TableVisBasic = memo(
+  ({
+    fireEvent,
+    table,
+    visConfig,
+    title,
+    uiStateProps: { columnsWidth, sort, setColumnsWidth, setSort },
+  }: TableVisBasicProps) => {
+    const { columns, rows } = useFormattedColumnsAndRows(table, visConfig);
+
+    // custom sorting is in place until the EuiDataGrid sorting gets rid of flaws -> https://github.com/elastic/eui/issues/4108
+    const sortedRows = useMemo(
+      () =>
+        sort.columnIndex !== null && sort.direction
+          ? orderBy(rows, columns[sort.columnIndex]?.id, sort.direction)
```

Comment:
Firstly, I don't think there could be boolean values at all, since they come from es response already as numbers (0 and 1).
Secondly, lodash works fine with boolean values if they are.
Don't think we'll see any difference in `orderBy` from lodash and `orderBy` filter from angular